mitodl/open-discussions
# OCW webhook

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #2612 
closes https://trello.com/c/AhXkk9oe/63-s3-triggered-webhook-to-import-update-ocw-courses

#### What's this PR do?
* Adds a new feature flag (`WEBHOOK_OCW`)
* Creates a webhook endpoint that will trigger an OCW import for a specific course based on the S3 key contained in the AWS notification JSON (if the new flag is `True`)
* The task to import the course should be delayed by `OCW_WEBHOOK_DELAY` seconds (as a precaution just in case it takes a little extra time for all relevant json files for a course to be written to the ocw bucket).
* The current daily task to iterate over all OCW courses won't do anything if the new flag is `True`
* The manual command `backpopulate_ocw_data` should still work because it sends an optional kwarg to override the feature flag.

#### How should this be manually tested?
* Add `FEATURE_WEBHOOK_OCW=True` to your .env file
* Via Postman or similar tool, post this data to the new endpoint `../api/v0/ocw_webhook/`:

```
{
    "Records": [
        {
            "s3": {
                "object": {
                    "key": "PROD/6/6.003/Fall_2011/6-003-signals-and-systems-fall-2011/0/1.json",
                }
            }
        }
    ]
}
```

* In the celery log you should soon see `Received task: course_catalog.tasks.get_ocw_courses`
* After a delay of at least 120 seconds, you should see `Task course_catalog.tasks.get_ocw_courses[<uid>] succeeded`
